IPs/RSKIPxxx.md

```diff
@@ -0,0 +1,262 @@
+
+## Internals
+
+In the following sections we specify how the Bridge must internally handle ordinals.
+
+## New Bridge Storage Address Mappings
+The utility internal function `getOrdinalIndex(btcTxHash, uint256 id)` returns the sequential index of the ordinal. The internal function g`etOrdinalIndexKey(uint index)` returns the key used to lookup the index. The key is computed as the keccak256 hash of the following string:
+
+`"OrdinalIndex-" + btcTxHash.toHexString() + id.toHexString()`
+
+Where "+" represents string concatenation and the toHexString() method returns the ASCII hexadecimal representation of the data without the "0x" prefix, using lowercase letters, and without trimming leading zeros . The btcTxHash hexadecimal string is always 64 bytes in length. The id hexadecimal string is always 64 bytes in length. No separator is used between these two hexadecimal strings.
+If the storage cell is empty, then the ordinal does not exist. This means that the address 0x00.. (20-byte all zeros) cannot own any ordinal.
+The cells contain an unsigned index. To lookup the ordinal metadata, a second storage key must be accessed. A parked ordinals is stored in a storage cels with an address that corresponds to the keccak256 hash of:
+
+`"Ordinal-" + uint64(index).toString() `
+
+Each cell contains the following fields:
+
+* **btcTxHash**: hash of original parking transaction
+* **outIndex**: Output index of the ordinal
+* **sats**: the BTC amount stored (which corresponds to the length of ordinals in sats).
+* **parkDate**: the parking date (changed on parking extension)
+* **owner**: the RSK address of the ordinal owner
+
+
+The methods that change the owner and parking date of the ordinal will modify the content of the index cell.
+
+The storage cell with an address equal to the hash of "OrdinalCount"stores the number of ordinals parked. Initially it is zero.
+
+## Single-use Ordinals Address Derivation
+The Powpeg ordinals address derivation starts with the creation of a custom redeem script that pushes the verOwner field. The verOwner field is defines as:
+
+* **version**: specifies the derivation version (1 byte) 
+* **owner**: ordinal owner address (a 20 byte RSK address)
+
+Therefore verOwner consist of a 21-byte binary chunk. The verOwner field is pushed and immediately dropped from the stack using the `OP_DROP` opcode, so the pushed data does not alter the processing of the redeem script.
+
+The redeem script for Powpeg-derived addresses is a modification of the standard Powpeg redeem script. We show how to build such scripts, but we omit the emergency multisig script path for clarity. The whole script must be considered when computing the address. The new custom redeem script has the following format:
```

Comment:
Just to be clear, the ordinal address redeem script will include the emergency multisig as well, right?